va/validation-authority.go

```diff
@@ -212,12 +208,10 @@ func (va *ValidationAuthorityImpl) fetchHTTP(identifier core.AcmeIdentifier, pat
 		scheme = "https"
 		port = va.httpsPort
 	}
-	portString := strconv.Itoa(port)
-	hostPort := net.JoinHostPort(host, portString)
 
 	url := &url.URL{
 		Scheme: scheme,
-		Host:   hostPort,
+		Host:   host,
```

Comment:
Ah. My impression was that the purpose of `va.httpPort` is to pretend we live in a world where the normal HTTP port is not 80, but say 50000 (e.g., we're testing and can't listen on a wkp). In that world, I would expect clients to leave off the :50000 because that's the implicit default.

But if I'm following you correctly, "http://foo:80/" should look like "http://foo/", and "http://foo:50000" should look like "http://foo:50000", even when `va.httpPort` is 50000. I.e., `va.httpPort` shouldn't change our definition of "normal" when it comes to URLs and Host headers. If that's right, I'll fix accordingly.